code/modules/black_market/illegalradio.dm

```diff
@@ -0,0 +1,370 @@
+#define MAIN 1
+#define SELLING 2
+#define DELIVERY 3
+
+var/list/global_illegal_radios = list()
+
+/proc/buzz_black_market() //Updates and buzzes all illegalradios so the screen is current.
+	for(var/obj/item/device/illegalradio/radio in global_illegal_radios)
+		if(radio.notifications && usr && radio in usr.contents)
```

Comment:
What are you using `usr` here ?

Reply:
There's no better way to get who's currently interacting with the radio unless I make a variable for the radio that gets set to usr, so, redundant step. The radio in usr.contents will prevent any bugged interactions.

Reply:
I'm still not buying this at all
Define a proc like this :

```dm
/obj/proc/get_holder()
      while (loc && !istype(loc, mob))
             return loc.get_holder()
       return loc
```
(on the off-chance that this doesn't exist already in some form).
Then, rather than `usr`, use `radio.get_holder()`.

Reply:
It's not redundancy, it's sanity. check if your usr is your user, at least.